internal/controller/predicates.go

```diff
@@ -3,31 +3,49 @@ package controller
 import (
 	"github.com/octopusdeploy/octopus-permissions-controller/internal/rules"
 	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
-func ServiceAccountDeletionPredicate() predicate.Predicate {
+func ManagedResourcePredicate() predicate.Predicate {
+	return predicate.NewPredicateFuncs(func(obj client.Object) bool {
+		return rules.IsOctopusManaged(obj.GetLabels())
+	})
+}
+
+func OwnedResourcePredicate() predicate.Predicate {
+	return predicate.Or(
+		ManagedResourcePredicate(),
+		predicate.Funcs{
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				return true
+			},
+		},
```

Comment:
Does this mean that we always process a deleted resource, even if we aren't managing it?

Reply:
Might need to catch up on this - feels like we've got a logic dupe for owned resources where we are checking `(Managed or Delete) and (GenerationChanged or Deleted)`

Reply:
Now unused